### ISSUE TITLE
Spycraft : The Great Game crashes

### ISSUE THREAD

**Issue by github-fanuser** [joncampbell123/dosbox-x#923]
I thought I would give DOSBox-x a try. I have been using DOSBox ECE for a while with no problems.
I like the GUI for conf settings and like that it uses the my entire laptop screen. I assume it is because DosBox-x supports Direct3d as an output option, ECE does not support Direct3d. 

The one problem I found is with Spycraft: The Great Game. It crashes without warning at the same point in the game. There is no error message that I have seen. I have tried different resolution modes (desktop, original and I also tried using my laptop screen resolution (1366x768) with the same result. I did not try the different output modes since that causes the display to be smaller and I would want to play it like that anyway. ECE does not cause the game to crash. I tried both the 64bit and 32bit version of DosBox-x.

**Comment by joncampbell123** [comment]
Do you mean DOSBox-X crashes, or the game crashes and hangs?

If it's DOSBox-X, try a different output driver, like OpenGL or surface.

**Comment by github-fanuser** [comment]
The game crashes. It (the game) crashes during the same point in the game.

**Comment by github-fanuser** [comment]
I thought would try your latest version to see if it fixes the problem with the game, Spycraft: The Great Game. Sadly, it does not. I realized that I never posted the error message which may or may not help you. 
Here is the error message.

Fatal Error (8) at line 1022
in ..\src\advres.c
Last resource 90000.TGA

That is the way the message appears on the screen (3 lines). Below the error message, is "C:\>" prompt. The error does not occur in the ECE edition of DosBox.   The error occurs during the same point in the game. This is the only game that causes the error. Thanks. Hope this helps.

**Comment by Allofich** [comment]
>  It crashes without warning at the same point in the game.

What point in the game?

**Comment by Allofich** [comment]
Does the crash still happen if you use

`allow low resolution vesa modes = false`

in your dosbox.conf file with the latest version of DOSBox-X?

**Comment by github-fanuser** [comment]
Yes. The crash still happens. It crashes during the scene when the
helicopter comes to pick you up in Tunisia. Dosbox is still running, the
game just exits. I am using the latest version 0.82.26. I have tried
previous versions of Dosbox-x. It has always crashed at this point in the
game.

UPDATE - I discovered by accident that if I launch DOSBox-x in a command
prompt and then run the game manually, the crash doesn't happen. I
originally ran the game from a Dos menu program (Direct Access 5.1). It's
weird because none of the other Dosbox flavors have this problem.

I am skeptical that the problem is gone. I thought I should post this
message anyway in case some has the game wants to try to recreate the
problem using the menu program. They'll be able to find the menu program on
Internet.

**Comment by github-fanuser** [comment]
Well the game appears to run correctly as long as I start the game from the command prompt and NOT the Direct Access menu program. Don't know why menu program causes problem with the game. This is the only flavor of DOSBox that has this problem. I have used the official version (0.74), ECE, and MB6; none of these have the problem. Maybe someone out there will discover the problem and post it so it can be fixed. For now, I will continue to run the game from the command prompt and hope the problem. I really like this version; it has a menu bar, more programs in the Z drive, the key mapper seems to have more features, and the screen size is bigger (no black bars). Thanks, Us there some documentation that explains all the features and the programs in the Z drive.

By the way, I am using the windows 32 bit SDL1 version. I thought this was the simplest version to test the game on.

**Comment by joncampbell123** [comment]
Do you run LOADFIX before starting the game?

**Comment by github-fanuser** [comment]
Yes. I tried that. Didn't make a difference.

**Comment by joncampbell123** [comment]
If the game does not require VESA BIOS extensions, does machine=vgaonly help?

**Comment by joncampbell123** [comment]
Just found a copy on GOG.com, will begin testing now.

**Comment by github-fanuser** [comment]
Ok thanks. I haven't had a chance to try your latest suggestion -
machine=vgaonly

You also need Direct Access 5.1 menu. They crash happens when the
helicopter picks you up from Tunisia; towards the end of the game.

**Comment by github-fanuser** [comment]
Just tested it. It does require VESA BIOS.

**Comment by joncampbell123** [comment]
That it does. This is the first game I've ever seen that uses 16-bit 5:5:5 RGB and FMV from AVI files. FFMPEG says the video codec is "Truemotion". Interesting.
![spydos_000](https://user-images.githubusercontent.com/6245486/74640768-f3d95300-5124-11ea-9af5-e56a3d94cb74.png)

**Comment by joncampbell123** [comment]
Oh please, not even today's OCR can resolve a license place like that XD
![spydos_001](https://user-images.githubusercontent.com/6245486/74641316-f0929700-5125-11ea-9a5c-e615101b1bf0.png)

**Comment by joncampbell123** [comment]
Please explain how to obtain the Direct Access Menu

**Comment by joncampbell123** [comment]
Did you know the AVI and WAV files in the DATA directory are playable using FFMPEG and VLC Player?

**Comment by github-fanuser** [comment]
https://winworldpc.com/download/46c2abc3-83c3-be18-c39a-11c3a4e284a2

Try the above link. Version 5.19 is the one I'm using.

**Comment by github-fanuser** [comment]
No. Is that helpful? I might be able to send you the saved game file where
the game crashes. I need to figure out what the filename; so you don't have
to play the whole game.

I appreciate you taking the time to check it out. I have no idea why the
menu program would cause the game to crash. I have not tried other menu
programs.

**Comment by joncampbell123** [comment]
Please post the save file. I have Direct Access installed in C:\DA51

**Comment by github-fanuser** [comment]
Here is the saved game. It will probably be in slot 5; titled "Heading  to
Tunisia". You will have to play the accomplish the mission first. The game
should crash and return to  the DA menu when the chopper comes in to pick
you up. If it does not crash, then you will be prompted to save the game.

Also, when you load the saved game, it will prompt you to save. You can
choose to save or cancel. It does not matter. I usually Cancel since it is
already saved.

I hope you are able to reproduce the problem. I tried to create a video to
send to you, but I could figure out how to get it to work. I use F11+v, but
couldn't find it.

**Comment by github-fanuser** [comment]
I haven't heard from you in several days. Did you get the saved game file?

**Comment by joncampbell123** [comment]
Sorry about the delay... where is the download link for the file?

**Comment by github-fanuser** [comment]
I sent it via replying using email.

**Comment by joncampbell123** [comment]
I saw that, but the email didn't appear to have anything attached.

**Comment by github-fanuser** [comment]
[Spycraft.zip](https://github.com/joncampbell123/dosbox-x/files/4238021/Spycraft.zip)

I am attempting to send the file again.

**Comment by joncampbell123** [comment]
@github-fanuser That worked, looks like the file made it.

**Comment by github-fanuser** [comment]
Great. Let me know how the testing goes.

**Comment by joncampbell123** [comment]
Nope. Chopper picked me up without crashing in game... and then later blew up.

**Comment by joncampbell123** [comment]
[dosbox.conf.zip](https://github.com/joncampbell123/dosbox-x/files/4238959/dosbox.conf.zip)

**Comment by joncampbell123** [comment]
Second round and it hung after the FMV showing the chopper landing.

**Comment by joncampbell123** [comment]
A quick look at where it's hung and the ASM suggests it's stuck in an endless loop following a linked list.

EAX=FFFFFFFF  ESI=00465008  DS=0168   ES=0168   FS=0168   GS=0168   SS=0168 Pr32
EBX=FFFFFFFB  EDI=FFFFFFFC  CS=0160   EIP=002F3D04  C1 Z0 S1 O0 A1 P1 D0 I1 T0
ECX=00465028  EBP=00398008  ds: b:00000000 type:12 parbg      NOPG  IOPL0  CPL0
EDX=BD2C67D7  ESP=00397FC0      l:FFFFFFFF dpl : 0 10011    2215836316

0160:002F3D04 39C8                cmp  eax,ecx
0160:002F3D06 740B                je   002F3D13 ($+b)         (no jmp)
0160:002F3D08 8B18                mov  ebx,[eax]
0160:002F3D0A 01FB                add  ebx,edi
0160:002F3D0C 8B4008              mov  eax,[eax+0008]
0160:002F3D0F 01DA                add  edx,ebx
0160:002F3D11 EBF1                jmp  short 002F3D04 ($-f)   (up)
0160:002F3D13 8B7608              mov  esi,[esi+0008]
0160:002F3D16 85F6                test esi,esi
0160:002F3D18 75E4                jne  002F3CFE ($-1c)        (up)
0160:002F3D1A 89D0                mov  eax,edx

**Comment by github-fanuser** [comment]
When you say hung, do you mean the game exited and return to the DA51 menu?

When the helicopter picked you up and then blew up, did you start the game
from the DA51 menu?

**Comment by joncampbell123** [comment]
I mean it hung in that the cursor changed to an hourglass and the game did not do anything more. As I posted earlier, the debugger showed it running in an infinite loop.

The helicopter blew up, then returned to the main menu, which is understandable since that's a failure condition (right?)

**Comment by joncampbell123** [comment]
So far skipping the FMV sequences with spacebar seems to avoid crashes.

**Comment by github-fanuser** [comment]
Yes. I believe something about the menu causes the crash. Not sure if it's
a dosbox problem or if it can be fixed.

If you mount the drives and run the game from the prompt, it runs fine.

**Comment by github-fanuser** [comment]
Huh... I haven't tried that.

**Comment by joncampbell123** [comment]
I do wonder if it's just memory corruption bugs in the game related to EMS or just bugs that weren't caught by release date.

**Comment by github-fanuser** [comment]
I just tried skipping the video and it still crashed and return to the
menu. I guess it maybe some weird setup with my system.

Oh well... Thanks for trying. I will just run this particular game from a
prompt.

**Comment by joncampbell123** [comment]
It also might be worth it to see what version you are using.

I obtained my copy from GOG dot com, which may differ from your copy.

**Comment by github-fanuser** [comment]
My copy is an original. I purchased years ago mid 90's. It's also weird
that
Dosbox-x is the only one that crashes the game.  I have used the official
version, Megabuild 6, and ECE. None crash the game.

**Comment by joncampbell123** [comment]
Ha, OK. You can avoid the exploding helicopter ending by booting up the laptop, downloading 40%, then cancelling the download and throwing the laptop out the door.

It seems the joke's on you if you think that progress bar will ever get beyond 40%.

**Comment by github-fanuser** [comment]
You don't even have to wait 40%. As soon as the option to cancel comes up,
do it and throw out. The files you need download first. You are in the home
stretch after the helicopter scene.

**Comment by github-fanuser** [comment]
Wanted to give you an update. I removed the menu program. It seems that was
the problem. Don't know why. Dosbox-x is the only version of Dosbox that
has the problem. I really like your version. It uses the entire screen...no
black bars on the side. Also has more programs on the Z drive. Is there
documentation that explains all the stuff on the Z drive?

**Comment by joncampbell123** [comment]
Some of the stuff was inherited from DOSBox Daum and other branches. A few were added as additional features provided by DOSBox-X.

There is an issue to track writing a user guide open right now.

https://github.com/joncampbell123/dosbox-x/issues/1412

**Comment by Wengier** [comment]
The DOSBox-X user guide I wrote is now available from the following wiki. You can find the documentation that explains all the stuff on the Z drive in it, especially under the section named “DOSBox-X’s Supported Commands”:

https://github.com/Wengier/dosbox-x-wiki/wiki

**Comment by github-fanuser** [comment]
Thank you. I'll be sure to check it out.

**Comment by github-fanuser** [comment]
Hey, I thought I would give you an update. I finally figured out a solution
to Spycraft crashing when starting from Direct Access. You use DOS32A.exe
when launching the game.

Now, that I can I use the menu program again. I would like to create a menu
option for the dos prompt that with the info box that comes up when you
first start dosbox. Is this possible? I tried command.com and I do get a
prompt, but no info box and intro.com does not work. Thank you!
